frontend/src/metabase/ui/components/inputs/MultiAutocomplete/MultiAutocomplete.tsx

```diff
@@ -1,21 +1,32 @@
-import type { MultiSelectProps, SelectItem } from "@mantine/core";
+import type { MultiSelectProps, SelectItemProps } from "@mantine/core";
 import { MultiSelect, Tooltip } from "@mantine/core";
 import { useUncontrolled } from "@mantine/hooks";
-import type { ClipboardEvent, FocusEvent } from "react";
-import { useMemo, useState } from "react";
+import type { ClipboardEvent, FocusEvent, ReactNode } from "react";
+import { useMemo, useState, forwardRef } from "react";
 import { t } from "ttag";
 
 import { color } from "metabase/lib/colors";
 import { Icon } from "metabase/ui";
 
+import type { Base, SelectItem, FilterFn } from "./types";
 import { parseValues, unique } from "./utils";
 
-export type MultiAutocompleteProps = Omit<MultiSelectProps, "shouldCreate"> & {
-  shouldCreate?: (query: string, selectedValues: string[]) => boolean;
+export type MultiAutocompleteProps<TValue extends Base> = Omit<
```

Comment:
Extra complexity shouldn't be pushed from dashboards here. The QB doesn't do that, for instance, but handles all the same cases.